cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/AnomalyState.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2019 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.detector;
+
+import com.linkedin.cruisecontrol.detector.Anomaly;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+
+/**
+ * A class to indicate how an anomaly is handled.
+ */
+public class AnomalyState {
+  private final Anomaly _anomaly;
+  private final long _detectionMs;
```

Comment:
`AnomalyState` is created by `AnomalyDetector` when it poll from the blocking queue and begin to handle the anomaly(it may not handle the anomaly after polling from the queue due to executor is busy), so if we record the anomaly detection time as the time we create `AnomalyState`,  it might be inaccurate.
I feel a better idea will be record anomaly detection time as the time anomaly is enqueue to the blocking queue by `GoalViolationDetector/BrokerFailureDetector/MetricAnomalyDetector`.

Reply:
I believe you mean "time anomaly is enqueue to the blocking queue" -> "time anomaly is polled from the blocking queue"?
If so, the new logic regarding moving the `addAnomalyDetectionState` before the corresponding check should resolve this.

Reply:
yes, the logic change mitigate this issue, now the time we are recording are late by `dequeue_time-enqueue_time` and I think it is ok.